ledger/participant-integration-api/src/main/scala/platform/store/dao/events/TransactionsTreeStreamReader.scala

```diff
@@ -0,0 +1,302 @@
+    Ordering.by[EventStorageBackend.Entry[Raw.TreeEvent], Long](_.eventSequentialId)
+
+  def streamTreeTransaction(
+      queryRange: EventsRange[(Offset, Long)],
+      requestingParties: Set[Party],
+      eventProjectionProperties: EventProjectionProperties,
+  )(implicit
+      loggingContext: LoggingContext
+  ): Source[(Offset, GetTransactionTreesResponse), NotUsed] = {
+    val startExclusiveOffset: Offset = queryRange.startExclusive._1
+    val endInclusiveOffset: Offset = queryRange.endInclusive._1
+    val span =
+      Telemetry.Transactions.createSpan(startExclusiveOffset, endInclusiveOffset)(
+        qualifiedNameOfCurrentFunc
+      )
+    logger.debug(
+      s"streamTreeTransaction($startExclusiveOffset, $endInclusiveOffset, $requestingParties, $eventProjectionProperties)"
+    )
+    val sourceOfTreeTransactions = doStreamTreeTransaction(
+      queryRange,
+      requestingParties,
+      eventProjectionProperties,
+    )
+    sourceOfTreeTransactions
+      .wireTap(_ match {
+        case (_, response) =>
+          response.transactions.foreach(txn =>
+            Spans.addEventToSpan(
+              telemetry.Event("transaction", TraceIdentifiers.fromTransactionTree(txn)),
+              span,
+            )
+          )
+      })
+      .watchTermination()(endSpanOnTermination(span))
+  }
+
+  private def doStreamTreeTransaction(
+      queryRange: EventsRange[(Offset, Long)],
+      requestingParties: Set[Party],
+      eventProjectionProperties: EventProjectionProperties,
+  )(implicit
+      loggingContext: LoggingContext
+  ): Source[(Offset, GetTransactionTreesResponse), NotUsed] = {
+    val (startExclusiveOffset, startExclusiveEventSequentialId): (Offset, Long) =
+      queryRange.startExclusive
+    val (endInclusiveOffset, endInclusiveEventSequentialId): (Offset, Long) =
+      queryRange.endInclusive
+    val createEventIdQueriesLimiter =
+      new QueueBasedConcurrencyLimiter(maxParallelIdCreateQueries, executionContext)
+    val consumingEventIdQueriesLimiter =
+      new QueueBasedConcurrencyLimiter(maxParallelIdConsumingQueries, executionContext)
+    val nonConsumingEventIdQueriesLimiter =
+      new QueueBasedConcurrencyLimiter(maxParallelIdNonConsumingQueries, executionContext)
+    val payloadQueriesLimiter =
+      new QueueBasedConcurrencyLimiter(maxParallelPayloadQueries, executionContext)
+    // TODO etq: Use dedicated filter type for TreeTransactions (only parties)
+    val decomposedFilters = requestingParties.iterator.map(party => Filter(party, None)).toVector
+    val idPageSizing = IdQueryConfiguration(
+      maxIdPageSize = maxIdsPerIdPage,
+      // The full set of ids for tree transactions is the union of three disjoint sets ids:
+      // 1) ids for consuming events, 2) ids for non-consuming events and 3) ids for create events.
+      // We assign a third of the working memory to each source.
+      idPageWorkingMemoryBytes = maxWorkingMemoryInBytesForIdPages / 3,
```

Comment:
TODO: idPageSizing and reasoning in this comments looks fishy.
We're attempting to assign a third of the total working memory for id fetching to each source if ids, but 
then we have actually have five tables that we query those ids from. What's each id table can be queried multiple times - once for each simple (decomposed) filter constraint

Reply:
For the record: changed from `maxWorkingMemoryInBytesForIdPages / 3` to `maxWorkingMemoryInBytesForIdPages / 5` because we fetch ids from 5 distinct tables.
(In general, the number of ids queries we need to do the next id to fetch is `#number-of-tables times #number-of-decomposed-filters`)